compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -1782,10 +1787,12 @@ impl SharedEmitterMain {
             match message {
                 Ok(SharedEmitterMessage::Diagnostic(diag)) => {
                     let handler = sess.diagnostic();
-                    let mut d = rustc_errors::Diagnostic::new(diag.lvl, &diag.msg);
+                    let mut d = rustc_errors::Diagnostic::new(diag.lvl, String::new());
```

Comment:
nit: we could probably change the constructor here so we can pass the `DiagnosticMessage` in directly

Reply:
Constructor already is defined as `fn new<M: Into<DiagnosticMessage>>(level: Level, message: M)`, however `diag.msg` is `Vec<(DiagnosticMessage, Style)>`, so here I think it would be better to have another constructor that can take this `Vec` - something like `fn new_with_messages(level: Level, messages: Vec<(DiagnosticMessage, Style)>)`

Reply:
Yeah, I think that makes sense.